official/resnet/keras/keras_benchmark.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executes Keras benchmarks and accuracy tests."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from absl import flags
+from absl.testing import flagsaver
+import tensorflow as tf  # pylint: disable=g-bad-import-order
+
+
+DATA_DIR = '/data/cifar10_data/cifar-10-batches-bin'
+MIN_TOP_1_ACCURACY = 0.925
+MAX_TOP_1_ACCURACY = 0.938
+
+FLAGS = flags.FLAGS
+
+
+class KerasBenchmark(object):
+  """Base benchmark class with methods to simplify testing."""
+  local_flags = None
+
+  def __init__(self, output_dir=None, default_flags=None):
+    self.oss_report_object = None
+    self.output_dir = output_dir
+    self.default_flags = {}
+    if default_flags:
+      self.default_flags = default_flags
+
+  def _get_model_dir(self, folder_name):
+    return os.path.join(self.output_dir, folder_name)
+
+  def _setup(self, flag_methods=None):
+    """Sets ups up and resets flags before each test.
+
+    Args:
+      flag_methods: list of methods to execute that add global flags.
+
+    """
+    tf.logging.set_verbosity(tf.logging.DEBUG)
+    if KerasBenchmark.local_flags is None:
+      if flag_methods:
```

Comment:
Added TODO.  This cannot be defaulted to {} because of
http://pylint-messages.wikidot.com/messages:w0102

Reply:
It is probably not a problem here because flag_methods is immutable in `_setup()`. Just want to clarify this. Feel free to choose the style you prefer :)

Reply:
Understood it still triggers a lint error at least in my linter so I would have to add a pytlink disable which seems more a pain than just playing along.